### PR TITLE
Modify CI

### PR DESCRIPTION
**Pull request by Milo123459**
This PR fixes the CI currently failing.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -88,6 +88,16 @@ jobs:
...
         if: matrix.toolchain == 'nightly'
         run: cargo clean; cargo update -Z minimal-versions; cargo check
+      - run: rustup component add clippy --toolchain nightly-x86_64-unknown-linux-gnu
+      - name: Lint
+        if: matrix.toolchain == 'nightly'
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --locked --all-targets --all-features --tests
+      - name: FMT
```
Comment (arqunis) [inline .github/workflows/ci.yml]:
```suggestion
      - name: Format
```

**Review by arqunis — CHANGES_REQUESTED** [review]

**Comment by Milo123459** [comment]
Opened upstream issue: https://github.com/rust-lang/rust-clippy/issues/7735

**Comment by Milo123459** [comment]
https://github.com/rust-lang/rust-clippy/issues/7712 has been closed, when it comes out on nightly this can be merged

**Comment by nickelc** [comment]
Hmm, i don't see how moving the lint jobs to test build matrix would have fixed the issues

**Comment by Milo123459** [comment]
Perhaps this  was the wrong name; this should just be some changes to the CI

**Comment by nickelc** [comment]
The clippy & format are separated because they run faster than the test build matrix and the clippy job also post annotations of the clippy warnings.

So, you get a faster response to PRs without interfering the test build matrix.

**Comment by Milo123459** [comment]
Oh

**Pull request closed without merging**